### PR TITLE
Fix code quality errors

### PR DESCRIPTION
**Pull request by arpitjain099**
I ran CodeQL scanning on the project and found the following 10 alerts.

image

I fixed 4 alerts and made this PR. Please review
image

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import pytest  # noqa
-from pytest import skip, raises  # noqa
```
Comment (arpitjain099) [inline test/common.py:2]: The libraries were imported and not used

image

File: tabulate/__init__.py [hunk 2]
```diff
@@ -1352,9 +1355,7 @@ def _normalize_tabular_data(tabular_data, headers, showindex="default"):
 
     try:
         bool(headers)
-        is_headers2bool_broken = False  # noqa
     except ValueError:  # numpy.ndarray, pandas.core.index.Index, ...
-        is_headers2bool_broken = True  # noqa
```
Comment (arpitjain099) [inline tabulate/__init__.py:1357]: image

Python: [Assignment statements](http://docs.python.org/2/reference/simple_stmts.html#assignment-statements).
Common Weakness Enumeration: [CWE-563](https://cwe.mitre.org/data/definitions/563.html).

**Review by arpitjain099 — COMMENTED** [review]
Python: [Assignment statements](http://docs.python.org/2/reference/simple_stmts.html#assignment-statements).
Common Weakness Enumeration: [CWE-563](https://cwe.mitre.org/data/definitions/563.html).

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: "CodeQL"
```
Comment (arpitjain099) [inline .github/workflows/codeql.yml:1]: Adding CodeQL scanning

File: tabulate/__init__.py [hunk 4]
```diff
@@ -823,7 +824,9 @@ def _isnumber_with_thousands_separator(string):
     """
     try:
         string = string.decode()
-    except (UnicodeDecodeError, AttributeError):
+    except (UnicodeDecodeError, AttributeError) as e:
```
Comment (arpitjain099) [inline tabulate/__init__.py:827]: image
Common Weakness Enumeration: [CWE-390](https://cwe.mitre.org/data/definitions/390.html).

**Comment by eliegoudout** [comment]
Hello @arpitjain099,
Sorry for asking you here, but do you have the rights to review / merge PRs? I think this repo is lacking activity recently and I would really like to have #221 merged for a project. It closes 11 issues as well as adding functionality.

Thank you in advance, and sorry for commenting somewhere unrelated.

All the best!

**Comment by arpitjain099** [comment]
@eliegoudout no I do not. I didn't know this repo is not being actively managed anymore. This repo is being used by a lot of devs (more than 5M downloads from PyPI). Surprised it's not being managed anymore.

**Comment by astanin** [comment]
Hi @arpitjain099 @eliegoudout 

I'm the project owner and maintainer. I do and intend to maintain this project going forward but it is a hobby project, and it takes a back seat after the day job, family, sport, and other hobbies. In practice, that means

0) I review and merge submissions approximately once or twice per year

1) Priority is on maintaining backwards compatibility and supporting new Python versions

2) Low priority stuff that might never be merged: code formatting, changing too many lines of code, new dependencies, introducing new workflows, integrating with yet another CI service.

3) New features: medium priority. I consider this project to be good enough as it is. It was certainly good for my needs. I try to avoid breaking old use cases.

4) Submitting pull requests is the best way to change/fix something in the project.

5) Small focused PRs are more likely to be merged. I might not have time to cherry-pick changes from a huge PR which may also conflict with something else.

6) Expect ~ 6 months lead time before it goes into a release.

@eliegoudout pull request #221 seems good, it doesn't break stuff as far as I can see. I'm not sure about printing warnings. I'd rather prefer it to do its best silently. There're some code formatting issues, but nothing that black cannot fix.

**Comment by arpitjain099** [comment]
@astanin thank you for the reply. I see that there are some merge conflicts now - let me resolve them and send the PR back to you for review.

**Pull request merged**